CharlieBackend.Business/Services/MentorService.cs

```diff
@@ -233,7 +233,27 @@ public async Task<Result<MentorDto>> DisableMentorAsync(long mentorId)
 
             if (!await _accountService.DisableAccountAsync(accountId.Value))
             {
-                return Result<MentorDto>.GetError(ErrorCode.NotFound, "This account is already disabled.");
+                return Result<MentorDto>.GetError(ErrorCode.Conflict, "This account is already disabled.");
+            }
+
+            return mentorDto;
+
+        }
+
+        public async Task<Result<MentorDto>> EnableMentorAsync(long mentorId)
+        {
+            var accountId = await GetAccountId(mentorId);
```

Comment:
* Use `MentorRepository.GetByIdAsync()` to get `MentorEntity` instead of getting account by id and making excessive database calls.
* Update same logic everywhere else.

Reply:
It turned out that front doest need to obtain Dto from these endpoints. Therefore problem is solved as Entities are not obtained from db.